_docs-sources/pipelines/security/secrets.md

```diff
@@ -0,0 +1,217 @@
+that provider. We recommend that whenever possible, use the same principles as we do with AWS, and leverage ephemeral
+credentials to authenticate with the provider, granting least privilege access to the resources required for the
+operation and avoid writing long lived credentials to disk.
+
+### Configuring Providers in Terragrunt
+
+For example, let's say you are configuring the [CloudFlare Terraform provider](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs).
+
+There are multiple methods of authentication supported by CloudFlare in this provider to allow you to make authenticated
+API calls to CloudFlare services. How should you configure your `terragrunt.hcl` file to authenticate with CloudFlare,
+and how should you manage the credentials required for Terragrunt to access these secrets safely?
+
+First, take a look at the provider generated by default in the root `terragrunt.hcl` file for AWS authentication:
+
+```hcl
+generate "provider" {
+  path      = "provider.tf"
+  if_exists = "overwrite_terragrunt"
+  contents  = <<EOF
+provider "aws" {
+  region = "${local.aws_region}"
+  # Only these AWS Account IDs may be operated on by this template
+  allowed_account_ids = ["${local.account_id}"]
+  # tags
+  default_tags {
+    tags = ${jsonencode(local.tags)}
+  }
+}
+EOF
+}
+```
+
+This provider block is generated on the fly whenever a `terragrunt` command is run, and provides the necessary
+configuration for the AWS provider to discover the credentials that have been made available to the environment by
+the [configure-aws-credentials](https://github.com/aws-actions/configure-aws-credentials) GitHub Action.
+
+No secrets are written to disk to support this, and secrets are discovered at runtime by the AWS provider.
+
+Looking at CloudFlare documentation, there are multiple methods of authenticating the CloudFlare provider, including the
+use of the [api_token](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs#api_key) field in the `provider` block, as shown in the documentation:
+
+```hcl
+generate "cloudflare_provider" {
+  path      = "cloudflare-provider.tf"
+  if_exists = "overwrite_terragrunt"
+  contents  = <<EOF
+provider "cloudflare" {
+  api_token = var.cloudflare_api_token
+}
+EOF
+}
+```
+
+Populating this `var.cloudflare_api_token` for the provider requires having a `variable "cloudflare_api_token" {}`
+block in a `.tf` file that is checked into the repository, and setting the `TF_VAR_cloudflare_api_token` environment
+variable set to the value of the CloudFlare API token. The easiest way to do this is to leverage the `inputs` value
+in `terragrunt.hcl` files to set the value of the `cloudflare_api_token` variable to the value
+of the `CLOUDFLARE_API_TOKEN`
+
+```hcl
+inputs = {
+  cloudflare_api_token = "${run_cmd("--terragrunt-non-interactive", "./fetch-cloudflare-api-token.sh")}"
```

Comment:
```
ERRO[0000]   on /projects/gruntwork/terragrunt-tests/terragrunt.hcl line 2, in locals: 
ERRO[0000]    2:   api_token = run_cmd("--terragrunt-non-interactive", "./fetch-cloudflare-api-token.sh") 
ERRO[0000]                                              
ERRO[0000] Call to function "run_cmd" failed: exec: "--terragrunt-non-interactive": executable file not found in $PATH.
 
ERRO[0000] /projects/gruntwork/terragrunt-tests/terragrunt.hcl:2,15-23: Error in function call; Call to function "run_cmd" failed: exec: "--terragrunt-non-interactive": executable file not found in $PATH. 

```

Reply:
works:
```
api_token = run_cmd("./fetch-cloudflare-api-token.sh", "--terragrunt-non-interactive")
```

Reply:
Thanks, I was going for `--terragrunt-quiet`.